[Documentation] Update doxygen for v20 and v21 game-related changes

garbear opened the pull request:
## Description

This PR adds documentation for new skinning features in v20 and v21.

The following changes are made (note that codedocs.xyz uses an older version of doxygen than our website):

* Improved formatting of RetroPlayer info labels ([RetroPlayer.VideoFilter](https://codedocs.xyz/garbear/xbmc/modules__infolabels_boolean_conditions.html#RetroPlayer_VideoFilter))
* Improved `gamewindow` documentation ([Game Control](https://codedocs.xyz/garbear/xbmc/Game_Control.html))
* Added `gamecontroller` documentation ([Game Controller](https://codedocs.xyz/garbear/xbmc/Game_Controller.html))
* Added `gamecontrollerlist` documentation ([Game Controller List](https://codedocs.xyz/garbear/xbmc/Game_Controller_List.html))
* Added/fixed game window IDs ([Window IDs](https://codedocs.xyz/garbear/xbmc/window_ids.html))

Closes https://github.com/xbmc/xbmc/issues/24033.

Replaces https://github.com/xbmc/xbmc/pull/24034.

## Motivation and context

@scott967 brought to my attention that documentation was missing for the stuff added in v20 and v21.

## How has this been tested?

Rendered and available at the above links.

## What is the effect on users?

* Improved game-related documentation for skinners

## Types of change
<!--- What type of change does your code introduce? Put an `x` with no space in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

garbear committed:
[Doxy] Improve documentation of "gamewindow" control

garbear committed:
[Doxy] Improve documentation of "gamecontroller" and "gamecontrollerlist"

enen92 commented on xbmc/guilib/WindowIDs.dox:
```diff
@@ -142,7 +146,15 @@ This page shows the window names, the window definition, the window ID and the s
 
 \section window_ids_rm Additional revision history
 
+\subsection modules_rm_windowids_v21 Kodi v20 (Nexus)
```
This section is for removals, additions are already identified on the table above

garbear replied:
K, removing

garbear committed:
[Doxy] Update Window IDs

enen92 reviewed (approved):
Thank you

garbear committed:
[Doxy] Improve formatting of RetroPlayer.* built-in options

garbear commented:
OK, I've updated the PR to enhance the documentation and formatting, fixed doxygen errors, and uploaded the docs to codedocs.xyz. See links in PR description.

HitcherUK commented on xbmc/games/controllers/guicontrols/GUIGameController.dox:
All the capital A's end up with a link to [https://codedocs.xyz/garbear/xbmc/classA.html](https://codedocs.xyz/garbear/xbmc/classA.html) on the docs.

garbear replied:
It's due to an old doxygen, when it's uploaded to docs.kodi.tv it'll be correct. See:

Bad A: https://codedocs.xyz/garbear/xbmc/skin_Button_control.html

Good A: https://xbmc.github.io/docs.kodi.tv/master/kodi-base/dc/df7/skin__button_control.html

HitcherUK commented on xbmc/games/controllers/guicontrols/GUIGameController.dox:
```diff
@@ -0,0 +1,120 @@
+console ports. A multitap connected to port <b>`/2`</b> would have the address
+<b>`/2/game.controller.snes.multitap`</b>.
+
+A SNES controller connected to port 1 on a multitap would then have the address
```
This A also links to [[A](https://codedocs.xyz/garbear/xbmc/classA.html) SNES controller]([A](https://codedocs.xyz/garbear/xbmc/classA.html) SNES controller)

HitcherUK commented on xbmc/cores/RetroPlayer/guicontrols/GUIGameControl.dox:
```diff
@@ -22,20 +22,62 @@ game is being played.
+| videofilter   | The filter to use, see \ref RetroPlayer_VideoFilter "RetroPlayer.VideoFilter" for supported values. If empty, the value of the \ref RetroPlayer_VideoFilter "RetroPlayer.VideoFilter" info label is used.
+| stretchmode   | The stretch mode, see \ref RetroPlayer_StretchMode "RetroPlayer.StretchMode" for supported values. If empty, the value of the \ref RetroPlayer_StretchMode "RetroPlayer.StretchMode" info label is used.
+| rotation      | The rotation, in degrees counter-clockwise, see \ref RetroPlayer_VideoRotation "RetroPlayer.VideoRotation" for supported values. If empty, the value of the \ref RetroPlayer_VideoRotation "RetroPlayer.VideoRotation" info label is used.
+| pixels        | A path to a v20 savestate, whose pixels will be rendered instead of the active game. A value of <b>`-`</b> will disable rendering of any pixels.
```
And these A's also link to https://codedocs.xyz/garbear/xbmc/classA.html

HitcherUK commented:
Will you also be updating https://github.com/xbmc/xbmc/blob/d92683f457dea63be8315661b6ea14f4ca074683/xbmc/guilib/WindowIDs.dox and https://github.com/xbmc/xbmc/blob/master/addons/resource.language.en_gb/resources/strings.po?

garbear commented:
@HitcherUK I believe I updated WindowIDs.dox in the most recent version of this PR?

And I'll update strings.po.

HitcherUK commented:
Good to go, thanks.

garbear commented:
@HitcherUK Strings.po PR: https://github.com/xbmc/xbmc/pull/24071

garbear commented:
Window IDs didn't generate correctly: https://xbmc.github.io/docs.kodi.tv/master/kodi-dev-kit/window_ids.html

Ideas?

garbear commented:
Maybe I need to end each table row with `<p></p>` like here? https://github.com/xbmc/xbmc/blob/master/xbmc/guilib/WindowIDs.dox#L145

@ksooo Are the p tags required?

garbear commented:
@Alwin Do you know how we can fix the window ID table? It cuts off after the first game window: https://xbmc.github.io/docs.kodi.tv/master/kodi-dev-kit/window_ids.html

scott967 commented:
Not sure how this gets done, but would like a "friendly name" for window 10149, WINDOW_DIALOG_PERIPHERALS so I can avoid 
```
<visible>Window.IsVisible(10149)</visible>
```
which works perfectly fine, but not as readable IMHO.
